File: review-service/ReviewService/src/main/java/com/wedding/reviewService/entity/Review.java
package com.wedding.reviewService.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "reviews")
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "Listing ID is required")
    private Long listingId;

    @NotNull(message = "User ID is required")
    private Long userId;

    @NotNull(message = "Rating is required")
    @Min(value = 1, message = "Rating must be at least 1")
    @Max(value = 5, message = "Rating must be at most 5")
    private Integer rating;

    @Column(columnDefinition = "TEXT")
    private String comment;

    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: frontend-app/Frontend/src/app/types/review.ts
export interface Review {
    id: number;
    listingId: number;
    userId: number;
    rating: number;
    comment: string;
    createdAt: string; // ISO string
    userName?: string; // Optional, for display purposes
}

export interface ReviewRequest {
    listingId: number;
    rating: number;
    comment?: string;
}
